store/local/job_spec_adapter.go

```diff
@@ -31,18 +33,19 @@ func init() {
 // Job are inputs from user to create a job
 // yaml representation of the job
 type Job struct {
-	Version      int    `yaml:"version,omitempty" validate:"min=1,max=100"`
-	Name         string `validate:"min=3,max=1024"`
-	Owner        string `yaml:"owner" validate:"min=3,max=1024"`
-	Description  string `yaml:"description,omitempty"`
-	Schedule     JobSchedule
-	Behavior     JobBehavior
-	Task         JobTask
-	Asset        map[string]string `yaml:"asset,omitempty"`
-	Labels       map[string]string `yaml:"labels,omitempty"`
-	Dependencies []JobDependency
-	Hooks        []JobHook
-	Metadata     JobSpecMetadata `yaml:"metadata,omitempty"`
+	Version              int    `yaml:"version,omitempty" validate:"min=1,max=100"`
+	Name                 string `validate:"min=3,max=1024"`
+	Owner                string `yaml:"owner" validate:"min=3,max=1024"`
+	Description          string `yaml:"description,omitempty"`
+	Schedule             JobSchedule
+	Behavior             JobBehavior
+	Task                 JobTask
+	Asset                map[string]string `yaml:"asset,omitempty"`
+	Labels               map[string]string `yaml:"labels,omitempty"`
+	Dependencies         []JobDependency
+	Hooks                []JobHook
+	Metadata             JobSpecMetadata    `yaml:"metadata,omitempty"`
+	ExternalDependencies ExternalDependency `yaml:"external-dependencies,omitempty"`
```

Comment:
Can you move this under JobDependency as discussed

Reply:
addressed under latest commit